### ISSUE TITLE
Upgrade to python3 - GPkit now supports python 3

### ISSUE THREAD

**Issue by bqpd** [UCLA-StarAI/NaCL#3]

**Comment by khosravipasha** [comment]
> (just dropping a note here that GPkit now supports python 3!)

Awesome Ned, thank you for the notice, this helps a lot, I will update the repo and use python3 when  I get the chance.

**Comment by khosravipasha** [comment]
This is mostly done now and ready in python3-refactor branch.
https://github.com/UCLA-StarAI/NaCL/tree/python3-refactor

Just need to port the last notebook about explanationabilty, other code functaionallity should be working.

**Comment by bqpd** [comment]
re: the readme at that link, GPkit will default to mosek solvers if they're installed at build time; you shouldn't need to manually specify!

the difference between `mosek_cli` and `mosek_conif` is that the former uses an outdated GP-specific solver, while the latter uses the exponential cone solver that they actively maintain. Presently they're about comparable in speed and stability, but the former certainly won't be getting any faster.

**Comment by khosravipasha** [comment]
Thanks for the clarification.  Is there some note that I can link to in the reamde? I found out about `mosek_conif` by chance.

I used  `mosek_cli` when I was running experiments a year or two ago. It was decent speed, For our problems I think most of the time was spent in GPKit. IIRC, for the biggest problem I tried processing it took 20min in gpkit and  10s seconds in mosek_cli. But gpkit is so awesome and easy to use that I never looked into alternatives. 

If I want to try bigger problems might need to do that.  Specially, now, I am getting segfaults for our bigger problems that used to work. Not actively workong on this, so I willl debug next time I have chance. Not sure If it fails before or after calling the optimizers (both mosek_cli and mosek_conif and cvxopt give me segfault so I guess I will try an older version of gpkit see what happens)

**Comment by bqpd** [comment]
yeah a lot of large problems are so structured that they spend waaay longer in object creation / memory management then in solving :sweat: c'est la vie en Python

Defending and all that has delayed me from releasing v1.0 which'll have the full mosek_conif description, but the github issue and PRs are here: https://github.com/convexengineering/gpkit/issues/1396

**Comment by khosravipasha** [comment]
Yeah, specially when you have to do recursion in python.
Congrats on defending. Great, I will try again when v1.0 is out.

**Comment by bqpd** [comment]
(we'll also have mixed-integer support in v1, as seen in https://github.com/convexengineering/gpkit/issues/414)
